dlt/extract/pipe.py

```diff
@@ -822,34 +682,58 @@ def _get_source_item(self) -> ResolvablePipeItem:
             # if too many new sources is added we switch to fifo not to exhaust them
             if (
                 self._next_item_mode == "fifo"
-                or (sources_count - self._initial_sources_count) >= self.max_parallel_items
+                or sources_count - self._initial_sources_count
```

Comment:
I know that it works but lack of parens here is scary :)

Reply:
I like the parens too :) I didn't meant to edit this line